src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -255,9 +264,14 @@ private function createFirewalls(array $config, ContainerBuilder $container)
         // add authentication providers to authentication manager
         $authenticationProviders = array_map(function ($id) {
             return new Reference($id);
-        }, array_values(array_unique($authenticationProviders)));
+        }, array_unique($authenticationProviders));
```

Comment:
Why the removal of `array_values` here? This variable is not used in any different way for the rest of the scope (method). It may not be needed, but perhaps to keep the diff small for this PR? There are a million different refactorings I'd like to do on this class (like remove all the by-reference stuff), but we should probably try to not do it now.

Reply:
The keys are now used in the `LazyGuardManager` (and it's service locator), so it had to be removed to not loose the keys.